src/Features/Core/Portable/InlineParameterNameHints/AbstractInlineParameterNameHintsService.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.InlineParameterNameHints
+{
+    internal abstract class AbstractInlineParameterNameHintsService : IInlineParameterNameHintsService
+    {
+        public async Task<IEnumerable<InlineParameterHint>> GetInlineParameterNameHintsAsync(Document document, TextSpan textSpan, CancellationToken cancellationToken)
+        {
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var spans = new List<InlineParameterHint>();
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var nodes = root.DescendantNodes(textSpan);
+            spans = AddAllParameterNameHintLocations(semanticModel, nodes, spans, cancellationToken);
+            return spans;
+        }
+
+        protected abstract List<InlineParameterHint> AddAllParameterNameHintLocations(
```

Comment:
don't really see why this is returning a List.  Either have it be IEnuemrable, or ImmutableArray.

Reply:
oh wait.  it's even weirder :)   you both pass in teh value to add to, and you return it out as a return value.  Pick one or the other :)

Reply:
fixed